src/models/event.js

```diff
@@ -759,6 +755,24 @@ utils.extend(module.exports.MatrixEvent.prototype, {
      * @param {MatrixEvent} newEvent the event with the replacing content.
      */
     makeReplaced(newEvent) {
+        if (this.isRedacted()) {
+            return;
+        }
+        // ignore previous replacements
+        if (this._replacingEvent && this._replacingEvent.getTs() > newEvent.getTs()) {
+            return;
+        }
+        if (newEvent.isBeingDecrypted()) {
```

Comment:
What about if the current event is still being decrypted (like when reapplying replacement from a cached sync)?

Reply:
yep, have that on my list, but as it will be similar to handling back pagination, I'm putting it off till then.